### PR TITLE
Dep updates + hsd v5.0.1

### PR DESCRIPTION
**Pull request by rithvikvibhu**
### Updated without changes:
- `css-loader`, `jsdom`, `mini-css-extract-plugin`, `sass-loader`, `sinon`, `tape`
- `@sentry/electron`, `bignumber.js`,` classnames`, `copy-to-clipboard`, `deep-equal`, `mixpanel`, `moment`, `node-polyfill-webpack-plugin`, `sass`, `semver`, `uuid`, `winston`
- `electron`: 19.0.7 -> 22.0.0 (along with `electron-builder` and `@electron/remote`)

### Updated with changes:
- `qrcode.react`: change in import syntax
- `babel`: conflicting plugin option
- `webpack`: replace deprecated onBeforeSetupMiddleware with setupMiddlewares
- `react`: 17.0.2 -> 18.2.0
  - putting "hot" back in hot-reloading, now changing react code doesn't refresh the whole app and changes are applied immediately on change! (deprecated `react-hot-reloader` -> `react-refresh`)
  - enzyme adapter update - updated for now, but when setting up testing, we should remove this completely and go with a better (read: actively worked on) testing solution like RTL
  - `connected-react-router` works with react 18, but hasn't updated peer dep so requires `--legacy-peer-deps`. Instead, I've forked and added that react 18 as a valid peer dependency. This is another package to move away from in the future since it's not maintained.
  - updated `react-redux` and `redux-thunk` - some files had weird import paths linking to internal files that no longer exist, changed to a regular imports.
  - removed `react-16-node-hanging-test-fix` - no longer required since it is fixed in core react

### hsd update
- v5.0.0 has a small wallet migration, but not adding any message since it adds negligible delay.
- Bid repair code updated to work with multiple blinds
- `hs-client`: v0.0.11 -> v0.0.13

### CODE REVIEW

File: app/background/node/service.js [hunk 1]
```diff
@@ -236,7 +236,7 @@ export class NodeService extends EventEmitter {
       noDns: this.noDns,
       listen: this.networkName === 'regtest', // improves remote rpc dev/testing
       chainMigrate: 3,
-      walletMigrate: 1,
+      walletMigrate: 2,
```
Comment (pinheadmz) [inline app/background/node/service.js:239]: 👍

**Comment by pinheadmz** [comment]
> not adding any message since it adds negligible delay.

Why not? if a user has, i dunno, 100 wallets or more it may take a minute?

**Comment by rithvikvibhu** [comment]
>Why not? if a user has, i dunno, 100 wallets or more it may take a minute?

Packaged and tested in Linux. Created 100 wallets in bob v1 with:
```sh
for i in {1..100}; do hsw-cli mkwallet $i; done
```

Then launched Bob packaged from this PR:
```
[I:2023-01-12T14:39:55Z] (wallet) Opening WalletDB...
[I:2023-01-12T14:39:55Z] (wallet-migrations) Database needs migration(s):
  - Account lookahead migration - Account lookahead now supports up to 2^32 - 1

[D:2023-01-12T14:39:55Z] (wallet-migrations) Last migration 1, last available migration: 2
[I:2023-01-12T14:39:55Z] (wallet-migrations) There are 1 migrations.
[I:2023-01-12T14:39:55Z] (wallet-migrations) Migration 2 in progress...
[I:2023-01-12T14:39:55Z] (wallet-migrations) Migration 2 is done.
[I:2023-01-12T14:39:55Z] (wallet) Added 40602 hashes to WalletDB filter.
[...snip...]
[I:2023-01-12T14:39:55Z] (wallet) Wallet opened (1).
[I:2023-01-12T14:39:55Z] (wallet) Wallet opened (2).
[...snip...]
[I:2023-01-12T14:39:55Z] (wallet) Wallet opened (100).
```

It took < 1 second :D

**Comment by rithvikvibhu** [comment]
Hey @nodech Ledger doesn't work anymore because of electron's memory cage ([blog post](https://www.electronjs.org/blog/v8-memory-cage)).

It has been fixed in `node-hid/node-hid` [v2.1.2](https://github.com/node-hid/node-hid/releases).
`hsd-ledger` depends on your fork `nodech/node-hid` [v2.1.1-hsd.0](https://github.com/nodech/node-hid/releases/tag/v2.1.1-hsd.0). Can you port over that fix (looks like it's just one line) to your fork and bump `hsd-ledger`?

**Pull request merged**